packages/internals/src/arity.js

```diff
@@ -0,0 +1,47 @@
+const toString = Object.prototype.toString;
+const fnToString = Function.prototype.toString;
+const reHostCtor = /^\[object .+?Constructor\]$/;
+const reNative = RegExp(
+  "^" +
+    String(toString)
+      .replace(/[.*+?^${}()|[\]\/\\]/g, "\\$&")
+      .replace(/toString|(function).*?(?=\\\()| for .+?(?=\\\])/g, "$1.*?") +
+    "$"
+);
+
```

Comment:
Can we maybe add some code documentation for these functions? It's not exactly trivial what this module is doing. At least for me😄